nemo_skills/inference/generate.py

```diff
@@ -457,6 +464,21 @@ def sync_loop(self, data):
                     self.dump_outputs(outputs, data_points_batch, fout)
                     data_points_batch = []
 
+
+    def get_llm_generations(self, requests_in_progress, generations):
```

Comment:
I looked at the changes to this file and they seem ok to me. But please run some more tests to double check that normal async generations work properly (e.g. no corruption to the order or anything like that) as this is a very core functionality that we really don't want to break, especially if it's a silent break that doesn't result in an error but causes data corruption

Reply:
I ran the model is both sync and async generations, and the generations look comparable to me, even if not exactly the same for all instances. The contamination results are also similar across generations.